### PR TITLE
[DW-278] Initial implementation of the summary page

### PR DESCRIPTION
**Pull request by giacomolm**

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-config/configuration/update/gdpr-transparency-facet.groovy [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+package org.hippoecm.frontend.plugins.cms.admin.updater
+
+import org.onehippo.repository.update.BaseNodeUpdateVisitor
+
+import javax.jcr.Node
+import javax.jcr.Value
+
+class GdprTransparencyFacet extends BaseNodeUpdateVisitor {
```
[the remark this answers is not in the document]
Comment (jacsamell) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/gdpr-transparency-facet.groovy]: Fair enough

File: cms/src/main/java/uk/nhs/digital/common/ExtractInitialLetterDerivedDataFunction.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import javax.jcr.*;
+
+public class ExtractInitialLetterDerivedDataFunction extends DerivedDataFunction {
+
+    private static final long serialVersionUID = 1;
+
+    @Override
+    public Map<String, Value[]> compute(Map<String, Value[]> parameters) {
+        if (!parameters.containsKey("title")) {
+            parameters.clear();
```
[the remark this answers is not in the document]
Comment (giacomolm) [reply cms/src/main/java/uk/nhs/digital/common/ExtractInitialLetterDerivedDataFunction.java]: I am just following the implementation of the parent class. If the title is not there, the rule is to clear the parameters field

File: cms/src/main/java/uk/nhs/digital/common/ExtractInitialLetterDerivedDataFunction.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            Value[] title = parameters.get("title");
+            try {
+                if (title.length > 0 && !title[0].getString().isEmpty()) {
+                    parameters.put("titleInitialLetter",
+                        new Value[]{
+                            this.getValueFactory().createValue(title[0].getString().substring(0,1).toUpperCase())
+                        });
+                }
+            } catch (ValueFormatException valueFormatException) {
+                parameters.clear();
```
[the remark this answers is not in the document]
Comment (giacomolm) [reply cms/src/main/java/uk/nhs/digital/common/ExtractInitialLetterDerivedDataFunction.java]: Same comments as above, if you want more detail I can give you more hints

File: cms/src/main/java/uk/nhs/digital/common/ExtractInitialLetterDerivedDataFunction.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacsamell) [reply cms/src/main/java/uk/nhs/digital/common/ExtractInitialLetterDerivedDataFunction.java]: Ok as long as you're happy with it

**Pull request merged**